main/tests/server/src/com/google/refine/expr/functions/strings/ParseUriTest.java

```diff
@@ -20,22 +20,23 @@
     @BeforeTest
     public void init() {
         logger = LoggerFactory.getLogger(this.getClass());
-        sampleUri = "https://www.openrefine.org:80/documentation#download";
+        sampleUri = "https://www.openrefine.org:80/documentation#download?format=xml&os=mac";
     }
 
     @Test
     public void testParseUriValidParams() {
         Object res = invoke("parseUri", sampleUri);
         HashMap<String, String> resMap = ParsingUtilities.mapper.convertValue(res, new TypeReference<>() {
         });
+
         Assert.assertNotNull(res);
         Assert.assertEquals(resMap.get("path"), "/documentation");
         Assert.assertEquals(resMap.get("host"), "www.openrefine.org");
         Assert.assertEquals(resMap.get("port"), "80");
         Assert.assertEquals(resMap.get("query"), "");
         Assert.assertEquals(resMap.get("fragment"), "download");
         Assert.assertEquals(resMap.get("scheme"), "https");
-
+        Assert.assertNotNull(resMap.get("query_params"));
```

Comment:
Could you validate the data value (`format=xml&os=mac`) and not just that there is data?

Reply:
oh yes I can do that!

Reply:
@elroykanye : just a comment, don't change the code, but here, you could have just compare the value of the whole string, since we don't manipulate it's composition.

Also, but I'm not asking it, we could return the data in such a way that we could do:
`parseUri("https://www.openrefine.org:80/documentation#download?format=xml&os=mac").search[0].key`
or
`parseUri("https://www.openrefine.org:80/documentation#download?format=xml&os=mac").search['format']`

Then, the more precise test case would be necessary! ;-)

Regards, Antoine

P.S. I'm not asking to do that, merely suggesting it could be done.